### PR TITLE
[Test] Add test suite for download.jl

### LINKED ISSUE #6: [Test] Add test suite for download.jl

### PR DESCRIPTION
**Pull request by zerefwayne**

### CODE REVIEW

File: src/download.jl [hunk 1]
```diff
@@ -7,9 +7,12 @@ Returns the URL of the gpkg dataset of the country
...
     if contains(dataid, "GADM")
-        provider, country_code = split(dataid, '_')
-
-        "https://biogeo.ucdavis.edu/data/gadm3.6/gpkg/gadm36_$(country_code)_gpkg.zip"
+        try
+            provider, country_code = split(dataid, '_')
+            "https://biogeo.ucdavis.edu/data/gadm3.6/gpkg/gadm36_$(country_code)_gpkg.zip"
+        catch err
+            @error "Invalid dataset format, please use GADM_<country_code>."
+        end
```
Comment (juliohm) [inline src/download.jl]: So this part can be rewritten with a Julia regex. The try-catch shouldn't be needed.

File: src/download.jl [hunk 2]
```diff
@@ -23,6 +26,10 @@ Registers and downloads the desired dataset.
 function download(dataset_code)
     dataset_url = dataurl(dataset_code)
 
+    if dataset_url === nothing
+        return
+    end
+
```
Comment (juliohm) [inline src/download.jl]: Not idiomatic Julia

File: test/download.jl [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    url = Geography.dataurl("GADMIND")
+    @test isnothing(url)
+
+    # dataurl returns nothing when code doesn't contain GADM
+    url = Geography.dataurl("IND")
+    @test isnothing(url)
+
+    # dataurl returns nothing on empty string
+    url = Geography.dataurl("")
+    @test isnothing(url)
```
Comment (juliohm) [inline test/download.jl]: Should throw error instead and use `@test_throws` to check that the error was thrown.

**Review by juliohm — CHANGES_REQUESTED** [review]

File: src/download.jl [hunk 4]
```diff
@@ -7,11 +7,16 @@ Returns the URL of the gpkg dataset of the country
...
-        provider, country_code = split(dataid, '_')
-
-        "https://biogeo.ucdavis.edu/data/gadm3.6/gpkg/gadm36_$(country_code)_gpkg.zip"
+        try
+            provider, country_code = split(dataid, '_')
+            "https://biogeo.ucdavis.edu/data/gadm3.6/gpkg/gadm36_$(country_code)_gpkg.zip"
+        catch err
+            @error "invalid dataset format, please use GADM_<country_code>."
+            throw(ArgumentError("invalid dataset format"))
+        end
```
Comment (juliohm) [inline src/download.jl]: As explained, this can be much simpler with a regex. Are you familiar with regex?

File: test/runtests.jl [hunk 5]
```diff
@@ -1,6 +1,12 @@
...
 
+testfiles = [
+    "download.jl"
+]
+
 @testset "Geography.jl" begin
-    # Write your tests here.
+for testfile in testfiles
+    include(testfile)
+  end
```
Comment (juliohm) [inline test/runtests.jl:12]: Incorrect alignment of `end`

**Review by juliohm — CHANGES_REQUESTED** [review]

File: src/download.jl [hunk 6]
```diff
@@ -20,21 +18,20 @@ end
...
+            dataid,
+            "GADM dataset for $dataid",
+            dataurl(dataid),
             post_fetch_method=DataDeps.unpack
         )
     )
 
     # downloading the dataset if not available
-    @info "Successfully downloaded the files: ", readdir(@datadep_str dataset_code; join=true)
+    @info "Successfully downloaded the files: ", readdir(@datadep_str dataid; join=true)
```
Comment (juliohm) [inline src/download.jl:35]: Quick question, doesn't DataDep already print some info message like this at the end?

**Review by juliohm — APPROVED** [review]

File: src/download.jl (hunk 6 above)
Comment (zerefwayne) [reply src/download.jl:35]: No Datadeps doesn't print anything once download finishes.

**Comment by zerefwayne** [comment]
Thank you for approving. :tada: Merging this.

**Pull request merged**